doc/sphinx/src/customization

```diff
@@ -0,0 +1,143 @@
+.. _customization:
+
+Customizing ``singularity-eos``
+================================
+
+Custom Variant
+---------------
+
+If you would like to create your own custom variant with additional
+models (or a subset of models), you may do so by using the
+``eos_variant`` class. For example,
+
+.. code-block:: cpp
+
+  #include <singularity-eos/eos.hpp>
+  using namespace singularity;
+  
+  using MyEOS_t = eos_variant<IdealGas, Gruneisen>;
+
+This will create a new type, ``MyEOS_t`` which contains only the
+``IdealGas`` and ``Gruneisen`` classes. (All of these live under the
+``singularity`` namespace.)
```

Comment:
Do we want to add anything here about the helper functions @dholladay00 created for assembling complex type lists? I'd be fine with a small blurb to just go look at `eos.hpp` for examples.